modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/SelectionControls.js

```diff
@@ -57,12 +74,30 @@ const SelectionControls = ({
 	const [selectAllButtonVisible, setSelectAllButtonVisible] = useState(
 		initialSelectAllButtonVisible
 	);
+	const [incompatibleSelection, setIncompatibleSelection] = useState(false);
 
 	const searchContainerRef = useRef();
 
 	const updateControls = ({bulkSelection, elements}) => {
 		const currentPageSelectedElementsCount = elements.currentPageSelectedElements.size();
 
+		const selectedElementNodes = elements.allSelectedElements.getDOMNodes();
+
+		const datasets = selectedElementNodes.map(
+			(node) => node.dataset.modelclassname
+		);
+
+		const selectedDocumentTypes = datasets.filter(
+			(dataset, index) => datasets.indexOf(dataset) === index
+		);
+
+		if (selectedDocumentTypes.length > 1) {
+			setIncompatibleSelection(true);
+		}
+		else {
+			setIncompatibleSelection(false);
+		}
+
```

Comment:
Does this need to be a state variable? I feel like this logic could be within `disableActionIfNeeded`, where we do have access to `event.elements`.

In the case it needs to be in a state variable, the naming is a bit misleading, since it's not always incompatible. It should be something like `multipleTypesSelected`.